cli/util/api.go

```diff
@@ -185,6 +230,77 @@ func GetFullNodeAPI(ctx *cli.Context) (v0api.FullNode, jsonrpc.ClientCloser, err
 	return client.NewFullNodeRPCV0(ctx.Context, addr, headers)
 }
 
+// Not thread safe
+func OnSingleNode(ctx context.Context) context.Context {
+	return context.WithValue(ctx, "retry-node", new(*int))
+}
+
+func FullNodeProxy[T api.FullNode](ins []T, outstr *api.FullNodeStruct) {
+	outs := api.GetInternalStructs(outstr)
+
+	var rins []reflect.Value
+	//peertoNode := make(map[peer.ID]reflect.Value)
+	for _, in := range ins {
+		rin := reflect.ValueOf(in)
+		rins = append(rins, rin)
+		//peertoNode[ins] = rin
+	}
+
+	for _, out := range outs {
+		rint := reflect.ValueOf(out).Elem()
+
+		for f := 0; f < rint.NumField(); f++ {
+			field := rint.Type().Field(f)
+
+			var fns []reflect.Value
+			for _, rin := range rins {
+				fns = append(fns, rin.MethodByName(field.Name))
+			}
+
+			rint.Field(f).Set(reflect.MakeFunc(field.Type, func(args []reflect.Value) (results []reflect.Value) {
+				errorsToRetry := []error{&jsonrpc.RPCConnectionError{}, &jsonrpc.ErrClient{}}
+				initialBackoff, err := time.ParseDuration("1s")
+				if err != nil {
+					return nil
+				}
+
+				ctx := args[0].Interface().(context.Context)
+
+				curr := -1
```

Comment:
Can we add a comment explaining this -1?

Reply:
This logic needs to be made better. Right now, we always call node 0 first. Ideally we want to call the leader node first but getting that info in here is a challenge